### PR TITLE
[Feature] Use new API for fetching users and conversation via qualified iDs

### PR DESCRIPTION
**Pull request by typfel**
## What's new in this PR?

- Update to use the new API for fetching and creating users introduced in https://github.com/wireapp/wire-ios-data-model/pull/1168

## Dependencies

Needs releases with:

- [x] https://github.com/wireapp/wire-ios-data-model/pull/1168
- [x] https://github.com/wireapp/wire-ios-request-strategy/pull/310

## Notes

Not all endpoints and events have been updated to include qualified IDs so these will implemented in a future PR.

### CODE REVIEW

**Review by johnxnguyen — APPROVED** [review]

File: Source/Synchronization/Strategies/TypingStrategy.swift [hunk 1]
```diff
@@ -227,9 +227,10 @@ public class TypingStrategy : AbstractRequestStrategy, TearDownCapable, ZMEventC
         
         guard let userID = event.senderUUID,
               let conversationID = event.conversationUUID,
-              let user = ZMUser(remoteID: userID, createIfNeeded: true, in: managedObjectContext),
-              let conversation = conversationsByID?[conversationID] ?? ZMConversation(remoteID: conversationID, createIfNeeded: true, in: managedObjectContext)
+              let user = ZMUser.fetch(with: userID, in: managedObjectContext)
```
Comment (David-Henner) [inline Source/Synchronization/Strategies/TypingStrategy.swift:230]: should you be using `fetchOrCreate` here? Since the API used previously had `createIfNeeded: true` ?

File: Source/Synchronization/Transcoders/ZMConversationTranscoder.swift [hunk 2]
```diff
@@ -319,8 +319,9 @@ extension ZMConversation {
     
     fileprivate func updateRoleFromEventPayload(_ payload: [String: Any], userId: UUID) {
         guard let roleName = payload["conversation_role"] as? String else { return }
-        
-        let user = ZMUser(remoteID: userId, createIfNeeded: true, in: self.managedObjectContext!)!
+
+        // TODO jacob extract domain
+        let user = ZMUser.fetchOrCreate(with: userId, domain: nil, in: self.managedObjectContext!)
```
Comment (David-Henner) [inline Source/Synchronization/Transcoders/ZMConversationTranscoder.swift:324]: Will this TODO be addressed in another PR?

Comment (typfel) [reply Source/Synchronization/Transcoders/ZMConversationTranscoder.swift:324]: Yes, this will be addressed in another PR

**Review by David-Henner — APPROVED** [review]
